python/tank/commands/setup_project_params.py

```diff
@@ -815,9 +815,12 @@ def __init__(self, config_uri, sg, log):
         self._roots_data = self._read_roots_file()
 
         # if there are more than zero storages defined, ensure one of them is the primary storage
-        if len(self._roots_data) > 0 and constants.PRIMARY_STORAGE_NAME not in self._roots_data:
-            raise TankError("Looks like your configuration does not have a primary storage. "
-                            "This is required. Please contact support for more info.")
+        if len(self._roots_data) > 1 and constants.PRIMARY_STORAGE_NAME not in self._roots_data:
```

Comment:
pleased update the comment on line 817 as it no longer reflects what the code is doing... I suggest adding an explanation *why* the logic now works the way it does as well.